### ISSUE TITLE
RStduio external data view window DataTables error

### ISSUE THREAD

**Issue by a1785487** [rstudio/rstudio#15482]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2024.09.1+394
    OS Version      : Win10
    R Version       : 4.4.2 (2024-10-31 ucrt)

This problem has only started after I updated my RStudio installation yesterday to the current release.

Start RStudio, which loads the current project.   Within the project there is a markdown file which is loading a dataset and undertaking some data exploration.  The environment is empty, but within the project a single tibble is shown in a secondary View window (detached from the main RStudio desktop window). 

Run the Markdown file.  Each time any tibble is changed, multiple dialogs pop up stating the same message:
"RStudio
DataTables warning: table id=rsGridData - Cannot reinitialise DataTable. For more information about this error, please see http://datatables.net/tn/3"
the code blocks until the dialog is dismissed, which then, in my case, almost immediately pops up again.

 This will continue to happen during the entire session, until you close the secondary Viewer window is closed.  IF you re-create the View(tibble) after the window is closed, the error seems to go away.

During development I will potentially have mulitple tibbles viewed in the same external viewer window.  I shouldn't have to rebuilt this viewer window each time I restart work on the this project.

- [Y ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ Y] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ NA] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ Y] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by lachlansimpson** [comment]
Seen in internal ticket 113418

**Comment by racing-focus** [comment]
May I ask, is there a solution to this.

It is annoying for an external table / data frame to refresh every time a line of code is run before this error appears.


```
DataTables warning: table id=rsGridData - Cannot reinitialise DataTable. For more information about this error, please see http://datatables.net/tn/3

```

Cheers,

**Comment by JordanWean** [comment]
I'd really love a solution to this. This problem often causes 20+ popups to hit okay on every time I change something. A typical day for me now involves hitting 'Ok' hundreds of times  or more unless I exit out of the table I'm looking at and then re-open after operations. It really is a workflow killer.

**Comment by gdmcdonald** [comment]
> This problem often causes 20+ popups to hit okay on every time I change something. 

Same here. It also prevents me closing the View window (of the datatable) generating the popups, as I have to click OK before I can click close, but as soon as you click ok another one pops up immediately. I suspect it may also be causing the more frequent crashing of RStudio I've had this week.

**Comment by kevinushey** [comment]
I think I can reproduce with the following example. First, use the viewer to view a "large" data set.

```R
library(nycflights13)
View(flights)
```

Then, "pop out" the data viewer window into its own window. Next, do something that purportedly mutates the data set, e.g.

```R
flights <- flights
```

You should see the window flash. However, from that point on, the window will flash after every console execution, regardless of the code. If those code executions happen too frequently, then you might see this.

<img width="277" alt="Image" src="https://github.com/user-attachments/assets/02061ff1-fa5c-4d82-863c-796eec2099dd" />

**Comment by ronblum** [comment]
Verified in RStudio Desktop Pro 2025.08.0-daily+223.pro2.

**Comment by nanobrad** [comment]
@kevinushey, thank you so much! I was able to verify this on the 223 Win10+ daily as well. This has been plaguing me for months.

For anyone else running into this issue--the viewer does not necessarily need to be popped out--I'm guessing that it depends more on dataset size and number of viewers you have open (not whether or not they are popped) based on the commit. One is more likely to be popping the viewers out when there are a lot of columns or a lot of viewers...

The Windows daily builds, including 223 can be found here:
https://dailies.rstudio.com/rstudio/cucumberleaf-sunflower/electron/windows/

**Comment by Nova-Scotia** [comment]
This is still present in current stable build, yes? I'm on RStudio 2025.05.1 Build 513 "Mariposa Orchid" Release (ab7c1bc7, 2025-06-01) for windows; this is the same as what's available at https://posit.co/download/rstudio-desktop/.

I've never downloaded the dev version of RStudio before - but this bug is making it extremely difficult to write code! Do I just go for a dev version of Cumberleaf sunflower?

**Comment by kevinushey** [comment]
That's right -- the daily builds for Cucumberleaf Sunflower are all collected at https://dailies.rstudio.com/rstudio/cucumberleaf-sunflower/; you can pick the installer appropriate for your platform.

The latest builds are more-or-less release candidates; we're hoping to prepare a new release of RStudio this month.
